src/FilterPluginManagerFactory.php

```diff
@@ -6,26 +6,22 @@
 
 use Interop\Container\ContainerInterface;
 use Laminas\ServiceManager\Config;
-use Laminas\ServiceManager\FactoryInterface;
-use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
 
 use function is_array;
 
 class FilterPluginManagerFactory implements FactoryInterface
 {
-    /**
-     * laminas-servicemanager v2 support for invocation options.
-     *
-     * @param array
-     */
-    protected $creationOptions;
-
     /**
      * {@inheritDoc}
      *
-     * @return FilterPluginManager
+     * @param string $requestedName
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
-    public function __invoke(ContainerInterface $container, $name, ?array $options = null)
+    public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null): FilterPluginManager
```

Comment:
This constitutes a BC break, as any extensions of this class (and it **is** non-final) would break unless they also defined the same return type hint.